apps/terminal/signal_handlers/session.py

```diff
@@ -16,4 +18,3 @@ def on_session_finished(sender, instance: Session, created, **kwargs):
         return
     # 清理一次可能因 task 未执行的缓存数据
     Session.unlock_session(instance.id)
-
```

Comment:
The provided code has no obvious differences from the reference version and should work as expected without any additional checks or optimizations needed. If you want to optimize it further, consider these general tips:

1. Remove redundant `if` conditions by using `and` or `or` statements based on your logic.
2. Minimize repetitive functions with a single purpose. For example, the `on_session_pre_save` and `on_session_finished` could potentially be combined into one function that updates both fields.

However, since you did not specify which parts of the code are causing issues or need improvement, I cannot assist with optimizing this particular snippet effectively.